itest/lnd_open_channel_test.go

```diff
@@ -877,6 +877,114 @@ func testFundingExpiryBlocksOnPending(ht *lntest.HarnessTest) {
 	ht.MineBlocksAndAssertNumTxes(1, 1)
 }
 
+// testConfirmationsUntilActiveOnPending verifies that as a channel status
+// transitions from pending to active, the ConfirmationsUntilActive field of the
+// PendingChannel decreases to 0 as expected.
+func testConfirmationsUntilActiveOnPending(ht *lntest.HarnessTest) {
+	var (
+		numConfs uint32         = 5
+		chanAmt  btcutil.Amount = 100000
+	)
+
+	// As we need to create a channel that requires more than 1 confirmation
+	// before it's open, with the current set of defaults, we'll need to
+	// create a new node instance.
+	alice := ht.NewNodeWithCoins("Alice", nil)
+	bob := ht.NewNode("Bob", []string{
+		fmt.Sprintf("--bitcoin.defaultchanconfs=%v", numConfs),
+	})
+
+	// Ensure Alice and Bob are connected.
+	ht.EnsureConnected(alice, bob)
+
+	// Alice initiates a channel opening to Bob.
+	param := lntest.OpenChannelParams{Amt: chanAmt}
+	ht.OpenChannelAssertPending(alice, bob, param)
+
+	// Both Alice and Bob have one pending open channel.
+	ht.AssertNumPendingOpenChannels(alice, 1)
+	ht.AssertNumPendingOpenChannels(bob, 1)
+
+	// Since the funding transaction is not confirmed yet,
+	// ConfirmationsUntilActive will always be numConfs, and confirmation
+	// height will be 0.
+	require.Equal(ht, numConfs, ht.AssertNumPendingOpenChannels(alice,
+		1)[0].ConfirmationsUntilActive)
+	require.Equal(ht, numConfs, ht.AssertNumPendingOpenChannels(bob, 1)[0].
+		ConfirmationsUntilActive)
+	require.Equal(
+		ht, uint32(0),
+		ht.AssertNumPendingOpenChannels(alice, 1)[0].ConfirmationHeight,
+	)
+	require.Equal(
+		ht, uint32(0),
+		ht.AssertNumPendingOpenChannels(bob, 1)[0].ConfirmationHeight,
+	)
+
+	// Mine the first block containing the funding transaction, This
+	// confirms the funding transaction but does not change the channel
```

Comment:
Nit: confirms the funding transaction but the channel should still remain pending ...